### PR TITLE
Changes from fork

### LINKED ISSUE #34: Using `should_call` passes wrong `runtime_self`
Running this test fails

```
import pytest

from flexmock import flexmock


class SomeClass():
    def __init__(self, value):
        self.value = value

    def return_value(self):
        return self.value


def use_some_class(value):
    some = SomeClass(value)
    return some.return_value()


class TestSomeClass():
    def test_some_class(self):
        flexmock(SomeClass).should_call('return_value').once()
        assert use_some_class(4) == 4
```

The error is: `AttributeError: type object 'SomeClass' has no attribute 'value'`

I investigated a little bit with a debugger and it seems as though the `runtime_self` and the `self` that gets passed on to the method when it is called is the class and not the instance.

### PR DESCRIPTION
**Pull request by ollipa**
This PR contains all the commits from the fork that we maintained before Github project was transferred to flexmock organization.

Closes #34

@bkabrda, I removed RPM specfile and packit conf because I was not sure if there is still a need to keep those files. Let me know if you wish to keep the files and I'll re-add them.

**Breaking changes**:
- Dropped support for Python 2.7, 3.4, 3.5.
- Removed unittest2 integration. unittest2 is not maintained anymore.
- Flexmock needs to be imported explicitly using `from flexmock import flexmock`. The hack that allowed importing flexmock directly using `import flexmock`, doesn't really work well linters and automatic completion. However, I wonder if we should still keep the hack for backwards compatibility.

### CODE REVIEW

**Comment by lachmanfrantisek** [comment]
>  I removed RPM specfile and packit conf because I was not sure if there is still a need to keep those files. Let me know if you wish to keep the files and I'll re-add them.

This is for the [packit-service](https://github.com/apps/packit-as-a-service) to ensure the package can be built/instaled/tested in RPM-based systems. (Spoiler: I am one of the developers...;)  It can also provide Copr repositories for a branch or releases as well as helping `flexmock` maintainer(s) in Fedora to get the new versions there. I don't want to push you to use it, but let me know if you want to get more from it.

Regarding the specfile, it [can be downloaded](https://packit.dev/docs/faq/#how-can-i-download-rpm-spec-file-if-it-is-not-part-of-upstream-repository) during the process if you do not want to store it in the git repository. I can send a PR if you want.

Nevertheless, thanks for taking care of `flexmock`, we use it a lot in our project.

**Comment by ollipa** [comment]
@lachmanfrantisek, in that case I'll put the files back :slightly_smiling_face: I was just not sure if they were still needed.

**Comment by lachmanfrantisek** [comment]
/packit build

note: I've approved the new project location in the meantime.

**Comment by lachmanfrantisek** [comment]
> @lachmanfrantisek, in that case I'll put the files back slightly_smiling_face I was just not sure if they were still needed.

Thanks..;) Let me know if you have any trouble with the service.

File: .github/workflows/ci.yml
Comment (christophe-riolo) [inline .github/workflows/ci.yml:50]: what's that line?

File: CHANGELOG.rst
Comment (christophe-riolo) [inline CHANGELOG.rst:31]: I'm still not sure we should drop 3.4 and 3.5 already

**Comment by ollipa** [comment]
I think the spec file needs some tweaking and it seems that need to bring back setup.py as well. I'll continue tomorrow and let you know if I need help.

**Review by christophe-riolo — COMMENTED** [review]
I need to test a couple of things but I somehow cannot install python versions with pyenv, I need to figure that out

File: .github/workflows/ci.yml
Comment (ollipa) [reply .github/workflows/ci.yml:50]: With this config poetry creates virtual environment in the current folder. Otherwise it would go to the user's home directory.

File: CHANGELOG.rst
Comment (ollipa) [reply CHANGELOG.rst:31]: Why do you think so? Python 3.4 reached end of life already two years ago and Python 3.5 almost one year. Even pytest has dropped support for Python 3.4. If someone still needs Python 3.4 or 3.5, they can keep using an older flexmock version.

If we keep supporting Python 3.4 & 3.5 we would need to remove type hints.

**Comment by ollipa** [comment]
@lachmanfrantisek, it seems that the new project location is not in the allowlist anymore for some reason.

**Comment by mfocko** [comment]
> it seems that the new project location is not in the allowlist anymore for some reason.

It should be fine now :) (it was missing `.git` suffix for repo-only, in case you wanted to include more repositories, don't hesitate to reach out)

could you please trigger it to check? (I bypass the namespace check as admin)

**Comment by ollipa** [comment]
@lachmanfrantisek, packit builds are successful now. I couldn't get the tests working because flexmock package is not installed in the runner when tests are executed. Is it possible to install the created package before executing tests? The old test script added current folder to PYTHONPATH so it wasn't actually testing the package itself.

**Comment by TomasTomecek** [comment]
@ollipa I'm gonna take a look and help you with tests. Will try to make those work in my fork now and let you know.

Edit: Oh, now I understand the ask, it's about running tests during the build. That workflow is deprecated to some extent - will try to port it to the flow you described: install the built RPM and run tests on it.

**Comment by TomasTomecek** [comment]
You can see in my PR that tests and linters are now running in two separate steps for the same set of targets: https://github.com/TomasTomecek/flexmock/pull/1

The failures are coming from mypy, not sure if they are false or true positives.

Let me know what you think. You're welcome to cherry-pick those comments and include them here - I can file a PR if you want me to.

**Comment by ollipa** [comment]
@TomasTomecek, thanks! I cherry-picked your commits. Mypy probably failed because either Mypy or Python version didn't match the one we have in our CI. I think it's better not to run linters in Fedora test farm since we already run them in our CI and type checking and linting doesn't affect the runtime behavior.

Is it be possible to move some packit related files to their own subfolder? Ideally it would be nice to have most of the files in their own folder with a README file that would explain what the files do and how they can be maintained.

**Comment by TomasTomecek** [comment]
@ollipa except for .packit.yaml, everything can be placed in a subdirectory - I'll try to do that today - will build those changes on top of the current PR HEAD.

**Comment by TomasTomecek** [comment]
@ollipa I have finally finished the move: https://github.com/TomasTomecek/flexmock/pull/3

please let me know if I should include more documentation how to work with the files/services; sadly two paths cannot be moved there: .fmf/version (requirement from the testing framework) and .packit.yaml (requirement from the CI system)

File: pyproject.toml
Comment (christophe-riolo) [inline pyproject.toml:73]: We could try to reduce those at some point, and use local disables, sparingly

**Review by christophe-riolo — APPROVED** [review]
Back from vacation, sorry for the delay in review; this looks good and we can make that release I think !

**Comment by ollipa** [comment]
@TomasTomecek, thanks I included your commit. Documentation looks good to me. We can always extend it later if needed.

@adarshk7, do you still want to take a look at the changes before I merge?

File: pyproject.toml
Comment (ollipa) [reply pyproject.toml:73]: Yeah lets do some cleanup later. There were quite many places where pylint was complaining so I just disabled some rules globally. I will make an issue about this.

Comment (ollipa) [reply pyproject.toml:73]: https://github.com/flexmock/flexmock/issues/47

**Review by adarshk7 — APPROVED** [review]
Looks good to me :rocket:

**Pull request merged**
